src/hotspot/share/runtime/serviceThread.cpp

```diff
@@ -196,6 +199,10 @@ void ServiceThread::service_thread_entry(JavaThread* jt, TRAPS) {
     if (jvmti_tagmap_work) {
       JvmtiTagMap::flush_all_object_free_events();
     }
+
+    if (omworldtable_work) {
+      LightweightSynchronizer::resize_table(jt);
+    }
```

Comment:
The monitor deflation thread seems like it would be a better place to do the OMWorld table resizing, because it also does cleaning that might be a response to monitor deflation.  The ServiceThread does a lot, and the OMWorld table might notice the stall.  The other thread is also a service thread in a way, so I think this could be moved there instead.

Reply:
And when we do add shrinking, there'll be interactions with things that the monitor deflation thread is doing.